### PR TITLE
fix: replace invalid utf-8 sequences in agent logs

### LINKED ISSUE #13433: Coder agent crashes when script log contains invalid UTF-8
I've had a mistake in my `~/personalize` script that I run from a `coder_script` resource, such that the contents of a `.tgz` downloaded with cURL were printed to stdout instead of written to a file. This resulted in the Coder agent crashing:

```
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]: 2024-06-01 21:18:55.222 [warn]  metadata: metadata collection timed out  key=cpu_usage  timeout=1s
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]: 2024-06-01 21:18:55.223 [warn]  metadata: metadata collection timed out  key=ram_usage  timeout=1s
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]: 2024-06-01 21:18:55.245 [warn]  run exited with error ...
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:     error= error in routine send logs:
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:                github.com/coder/coder/v2/agent.(*apiConnRoutineManager).start.func1
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:                    /home/runner/actions-runner/_work/coder/coder/agent/agent.go:2025
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:              - failed to upload logs:
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:                github.com/coder/coder/v2/codersdk/agentsdk.(*LogSender).SendLoop
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:                    /home/runner/actions-runner/_work/coder/coder/codersdk/agentsdk/logs.go:433
Jun 01 21:18:55 coder-niklas-test coder-agent[9126]:              - string field contains invalid UTF-8
```

This is sub-par for the user experience and maintenance, as a crashing agent means no way for the user to get into the workspace and fix the script.

### PR DESCRIPTION
**Pull request by kylecarbs**
Fixes #13433.

### CODE REVIEW

**Review by spikecurtis — COMMENTED** [review]
This should have a regression test

**Comment by MrPeacockNLB** [comment]
Is it possible to backport this to stable branch? We are facincing this error in production too. Coder Version **v2.10.3**

File: codersdk/agentsdk/convert.go [hunk 1]
```diff
@@ -348,7 +348,7 @@ func ProtoFromLog(log Log) (*proto.Log, error) {
 	}
 	return &proto.Log{
 		CreatedAt: timestamppb.New(log.CreatedAt),
-		Output:    log.Output,
+		Output:    strings.ToValidUTF8(log.Output, ""),
```
Comment (coadler) [inline codersdk/agentsdk/convert.go:351]: I think we should probably replace invalid utf-8 with something, so it's obvious data is being deleted.

Comment (spikecurtis) [reply codersdk/agentsdk/convert.go:351]: Should we get cute and replace with something like ❌ or 😞 or 🤬, or just something like `<<invalid UTF-8>>`?

**Comment by spikecurtis** [comment]
Decided to just be bold and add the regression test and replacement with ❌ --- @coadler @kylecarbs revert me if you disagree.

**Review by coadler — APPROVED** [review]

**Comment by spikecurtis** [comment]
> Is it possible to backport this to stable branch? We are facing this error in production too. Coder Version **v2.10.3**

@MrPeacockNLB I think our policy for fixes unrelated to security is to backport to stable, which is 2.11 as of today.

**Pull request merged**